doc/iterations/2024.3/shapes/confirmation_dialog_on_delete.adoc

```diff
@@ -0,0 +1,52 @@
+= Confirmation dialog on delete
+
+== Problem
+
+Sirius Web does not have an undo/redo mechanism.
+When an element has been deleted, either from the Explorer view, from any representation, or any other view, there is no way to retrieve it.
+All delete actions are executable in one click.
+Any accidental click on a delete action can lead to the unwanted loss of data.
+
+== Key Result
+
+No delete action should be done in one click.
+
+== Solution
+
+A confirmation dialog should be displayed when a delete action (click on the delete tool on a node, click on delete key on a node, click on the delete menu item on an object in the Explorer view) has been called.
+This dialog should avoid or drastically minimized unwanted loss of data.
+
+As a specifier, It should be possible to enable/disable this capability on every representation description (Explorer view, any diagram description...).
+
+=== Diagram tools
+
+To go further on the subject, the confirmation dialog mechanism could be extended to any tool in diagrams, not only the delete tool.
+Thus, a specifier could enable/disable a confirmation dialog on the execution of its tools.
+On a tool description, the specifier could set a message expression and a button label expression (i.e. AQL expressions), allowing to call the confirmation dialog when the tool is executed by users.
+
+== Scenario
+
+=== Breadboarding
+
+The confirmation dialog could look like this:
+
+image:images/delete_dialog/delete_dialog.svg[Delete dialog]
+
+No cancel button will be provided, to be consistent with other Sirius Web dialogs.
+A click outside the confirmation dialog will cancel the deletion.
+
+For the specific cas of diagrams tools, the confirmation dialog will be the same, adapted with the message and button label set by the specifier.
+
+=== Cutting backs
```

Comment:
You should be able also to see the consequences of the deletion (x elements will be deleted. Are your sure?)